### PR TITLE
Add race and ethnicity bindings to bindings collection

### PR DESCRIPTION
**Pull request by bryantaustin13**
Issue #433 QI Core Value Set and Code System Query Issue
**Description**

visitExtensions was added to StructureDefinitionElementBindingVisitor to visit the extensions and collect the bindings especially for patient.extension:race and patient.extension:ethnicity.

- Github Issue:  
- [X ] I've read the contribution guidelines 
- [ X] Code compiles without errors
- [ ] Tests are created / updated
- [ ] Documentation is created / updated

By creating this PR you acknowledge that your contribution will be licensed under Apache 2.0

### CODE REVIEW

**Review by c-schuler — CHANGES_REQUESTED** [review]
I am not able to test anything in this PR, so I am unable to verify functionality. A few small changes requested and some comments.

Nice work!

File: tooling/pom.xml [hunk 1]
```diff
@@ -333,6 +333,12 @@
             <artifactId>mockserver-client-java</artifactId>
             <version>5.14.0</version>
         </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (c-schuler) [reply tooling/pom.xml]: Applied

File: tooling/src/main/java/org/opencds/cqf/tooling/acceleratorkit/StructureDefinitionElementBindingVisitor.java [hunk 2]
```diff
@@ -121,10 +121,84 @@ private void getBindings(String sdName, List<ElementDefinition> eds, String sdUR
...
+                    } else if (null != this.canonicalResourceDependenciesAtlas.getValueSets().getByCanonicalUrlWithVersion(sdbo.getBindingValueSetURL())) {
+                        valueSetVersion = this.canonicalResourceDependenciesAtlas.getValueSets().getByCanonicalUrlWithVersion(sdbo.getBindingValueSetURL()).getVersion();
+                        sdbo.setBindingValueSetName(this.canonicalResourceDependenciesAtlas.getValueSets().getByCanonicalUrlWithVersion(sdbo.getBindingValueSetURL()).getName());
+                        elementValueSet = this.canonicalResourceDependenciesAtlas.getValueSets().getByCanonicalUrlWithVersion(sdbo.getBindingValueSetURL());
+                    } else if (valueSetVersion.length() < 1 && bindingValueSet.contains("|")) {
+                        valueSetVersion = pipeVersion;
+                    }
+                    sdbo.setBindingValueSetVersion(valueSetVersion);
+                    if (null != elementValueSet) {
+                        StringBuilder codeSystemURLs = new StringBuilder();;
```
[the remark this answers is not in the document]
Comment (c-schuler) [reply tooling/src/main/java/org/opencds/cqf/tooling/acceleratorkit/StructureDefinitionElementBindingVisitor.java]: Applied

File: tooling/src/main/java/org/opencds/cqf/tooling/acceleratorkit/StructureDefinitionElementBindingVisitor.java [hunk 3]
```diff
@@ -121,10 +121,84 @@ private void getBindings(String sdName, List<ElementDefinition> eds, String sdUR
                 sdbo.setBindingValueSetVersion(valueSetVersion);
                 bindingObjects.put(sdName + "." + sdbo.getElementId(), sdbo);
             }
+            else if(ed.getId().equalsIgnoreCase("patient.extension:race")
```
[the remark this answers is not in the document]
Comment (c-schuler) [reply tooling/src/main/java/org/opencds/cqf/tooling/acceleratorkit/StructureDefinitionElementBindingVisitor.java]: Applied

File: tooling/src/main/java/org/opencds/cqf/tooling/acceleratorkit/StructureDefinitionElementBindingVisitor.java [hunk 4]
```diff
@@ -121,10 +121,84 @@ private void getBindings(String sdName, List<ElementDefinition> eds, String sdUR
...
+                        elementValueSet = this.canonicalResourceDependenciesAtlas.getValueSets().getByCanonicalUrlWithVersion(sdbo.getBindingValueSetURL());
+                    } else if (valueSetVersion.length() < 1 && bindingValueSet.contains("|")) {
+                        valueSetVersion = pipeVersion;
+                    }
+                    sdbo.setBindingValueSetVersion(valueSetVersion);
+                    if (null != elementValueSet) {
+                        StringBuilder codeSystemURLs = new StringBuilder();;
+                        Map<String, String> codeSystemsMap = new HashMap<>();
+                        getValueSetCodeSystems(elementValueSet, codeSystemsMap);
+                        if(null != codeSystemsMap && !codeSystemsMap.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (c-schuler) [reply tooling/src/main/java/org/opencds/cqf/tooling/acceleratorkit/StructureDefinitionElementBindingVisitor.java]: Applied

File: tooling/src/main/java/org/opencds/cqf/tooling/acceleratorkit/StructureDefinitionElementBindingVisitor.java [hunk 5]
```diff
@@ -121,10 +121,84 @@ private void getBindings(String sdName, List<ElementDefinition> eds, String sdUR
...
+            sdbo.setMustSupport("N");
+        }
+        if(ed.hasMin()){
+            String edCardinality = ed.getMin() + "..." + ed.getMax();
+            sdbo.setCardinality(edCardinality);
+        }
+        CanonicalType canonicalType = new CanonicalType();
+        canonicalType.setValue(String.valueOf(ed.getType().get(0).getProfile().get(0)));
+        ed.getType().get(0).getProfile().get(0).getValueAsString();
+        String urlValue = ed.getType().get(0).getProfile().get(0).toString();
```
[the remark this answers is not in the document]
Comment (c-schuler) [reply tooling/src/main/java/org/opencds/cqf/tooling/acceleratorkit/StructureDefinitionElementBindingVisitor.java]: Applied

**Review by c-schuler — APPROVED** [review]
Looks good! Waiting on @brynrhodes to give the final go-ahead.

**Review by brynrhodes — APPROVED** [review]

**Pull request merged**